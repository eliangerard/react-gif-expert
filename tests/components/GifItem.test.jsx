import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

let title = 'Curso';
let url = 'https://www.udemy.com/course/react-cero-experto/learn/lecture/19784000#overview';

describe('GifItem component tests', ()=> {
    test('Should match with snapshot', ()=> {
        const {container} = render(<GifItem title= {title} url={url}/>);
        expect(container).toMatchSnapshot();
    });
    test('debe mostrar la imagen con el url y el alt indicado', () => {
        render( <GifItem title={title} url = {url}/>);
        //expect(screen.getByRole('img').src).toBe(url);

        const {src, alt} = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    });
    test('debe mostrar el titulo del componente', () => {
        render(<GifItem title = {title} url={url}/>);
        expect( screen.getByText(title)).toBeTruthy();
    })
})