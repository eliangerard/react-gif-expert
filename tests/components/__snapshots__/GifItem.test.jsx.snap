// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem component tests Should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Curso"
      src="https://www.udemy.com/course/react-cero-experto/learn/lecture/19784000#overview"
    />
    <p>
      Curso
    </p>
  </div>
</div>
`;
